### ISSUE TITLE
MS SQL Server Support preparation: targeted tests

### ISSUE THREAD

**Issue by Steve-Mcl** [FlowFuse/flowfuse#3696]
In the discovery made during the POC carried out as part of https://github.com/FlowFuse/flowfuse/issues/3689, some tables cascades need to be removed to prevent errors around self-referencing, cycles and multiple cascade paths. For this, I have added a new task that should be addressed and merged before any other. This task "targeted tests" should explicitly test the cascades and any other DB changes proposed in the branch developed for https://github.com/FlowFuse/flowfuse/issues/3689. No MSSQL specific changes should be merged until these targeted tests are merged and passing for PG and SQLite.

**Comment by Steve-Mcl** [comment]
Acceptance Criteria: the below tables cascades will need to be modified and should therefore have explicit tests before merging any changes for MSSQL 

- [ ] ProjectStacks
- [ ] ProjectTypes
- [ ] Projects
- [ ] DeviceGroups
- [ ] Devices
- [ ] ProjectSnapshots
- [ ] PipelineStages

**Comment by Steve-Mcl** [comment]
Just additional tests - no application logic was changed.  Moving to "Done"
